azure/Guidelines.md

```diff
@@ -292,11 +292,15 @@ There are 2 kinds of errors:
 
 :white_check_mark: **DO** return an `x-ms-error-code` response header with a string error code indicating what went wrong.
 
-*NOTE: Error code values are part of your API contract (because customer code is likely to do comparisons against them) and cannot change in the future.*
+*NOTE: `x-ms-error-code` values are part of your API contract (because customer code is likely to do comparisons against them) and cannot change in the future.*
 
-:white_check_mark: **DO** carefully craft `x-ms-error-code` string values for errors that are recoverable at runtime.
+:heavy_check_mark: **YOU MAY** implement the `x-ms-error-code` values as an enum with `"modelAsString": true` because it's possible add new values over time.  In particular, it's only a breaking change if the same conditions result in a different top-level error code.
```

Comment:
```suggestion
:heavy_check_mark: **YOU MAY** implement the `x-ms-error-code` values as an enum with `"modelAsString": true` because it's possible add new values over time.  In particular, it's only a breaking change if the same conditions result in a *different* top-level error code.
```

Reply:
I think it would be nice to separate the error concepts from the swagger suggestions. The service implementors need to know what's in the contract and how to craft an error. This is not related to swagger.
The swagger is how to describe the implementation and explaining these rules is useful but I think should be separated out because different people are likely to care about these 2 things differently.

Reply:
That's a fair suggestion, but I'd still say if the service designers aren't thinking about how they'll eventually do this in the swagger then we'll be told it's an impossible amount of work to pull the list of errors out of the service's front end.  Would you separate this out into a separate guideline?  Or remove from the document entirely?

Reply:
@JeffreyRichter -- We already talk about `Extensible Enums` in the spec, see: [Extensible Enums](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#use-extensible-enums). These are referenced in a number of places. We did this, in part, because at the time, we didn't have the [Open API Style Guidelines](https://aka.ms/azapi/style). What I'd like to do is be consistent within the Guidelines that we have now, and point to the extensible enum section. We will open a new issue to determine if we should refactor the guidelines to remove the section to the style guide--which we can do on a separate pass.